### DISCUSSION TITLE
How to use curl_multi_poll() with extra file descriptors

### DISCUSSION THREAD

**Discussion by synchronicma** [curl/curl#13835] [Q&A] [answered]
Hello everybody,
I use libcurl and I use select() for my file descriptors and curl_multi_poll() for cURL descriptors.
I would like manage my file descriptors by curl_multi_poll(), but I don't know how to do it.
All example in curl.se use NULL for "extra_fds" parameter.
I don't find any example :(

In my case, I would like read a FIFO to get the next request to pass to curl.

Please help me.

**Comment by synchronicma** [comment]
While waiting to combine the two functions, here is a complete example of my program attached
[curl_test_async_and_fifo.c.txt](https://github.com/user-attachments/files/15511225/curl_test_async_and_fifo.c.txt)

**Comment by bagder** [comment, marked as answer]
I made this PR to expand the example in the man page to also wait on a private file descriptor. Does it help?

#13842

**Reply by synchronicma** [reply]
Thank you.
It's a big help
